Build failure: Unknown scheme (dart-ext) for dart-ext:src/serial_port

_From @tiholic on March 31, 2018 21:48_

## Steps to Reproduce

```dart
import 'package:flutter/material.dart';
import 'package:serial_port/serial_port.dart';

void main() => runApp(new MyApp());
//...
```

## Logs

`futter run`
```
Running "flutter packages get" in flutter_app...
Launching lib\main.dart on ONEPLUS A5010 in debug mode...
Initializing gradle...
Resolving dependencies...
Running 'gradlew assembleDebug'...
Error: Unhandled exception:
Could not load "dart-ext:src/serial_port": Unknown scheme (dart-ext) for dart-ext:src/serial_port
null'package:serial_port/serial_port.dart': error: line 20 pos 1: library handler failed
import 'dart-ext:src/serial_port';
^
'file:///D:/Workspace/Flutter/workspace/flutter_app/lib/main.dart': error: line 2 pos 1: library handler failed
import 'package:serial_port/serial_port.dart';
^
Failed to run the Flutter compiler. Exit code: 254

FAILURE: Build failed with an exception.

* Where:
Script 'D:\Workspace\Flutter\flutter\packages\flutter_tools\gradle\flutter.gradle' line: 421

* What went wrong:
Execution failed for task ':app:flutterBuildDebug'.
> Process 'command 'D:\Workspace\Flutter\flutter\bin\flutter.bat'' finished with non-zero exit value 254

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

* Get more help at https://help.gradle.org

BUILD FAILED in 13s
Finished with error: Gradle build failed: 1
```

`flutter analyze`

```
Analyzing D:\Workspace\Flutter\workspace\flutter_app...
No issues found!
Ran in 13.4s

```

## Flutter Doctor

`flutter doctor -v`
```
[√] Flutter (Channel beta, v0.1.5, on Microsoft Windows [Version 10.0.16299.309], locale en-IN)
    • Flutter version 0.1.5 at D:\Workspace\Flutter\flutter
    • Framework revision 3ea4d06340 (5 weeks ago), 2018-02-22 11:12:39 -0800
    • Engine revision ead227f118
    • Dart version 2.0.0-dev.28.0.flutter-0b4f01f759

[√] Android toolchain - develop for Android devices (Android SDK 26.0.2)
    • Android SDK at C:\Users\Rohit\AppData\Local\Android\sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-27, build-tools 26.0.2
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-915-b01)

[√] Android Studio (version 3.0)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-915-b01)

[√] Connected devices (1 available)
    • ONEPLUS A5010 • f0f79579 • android-arm64 • Android 8.0.0 (API 26)

• No issues found!
```

_Copied from original issue: flutter/flutter#16133_

--- a/.travis.yml
+++ b/.travis.yml
@@ -5,7 +5,7 @@ language: dart
 
 dart:
   # Keep version in-sync with Dockerfile and app/lib/shared/versions.dart
-  - 2.0.0-dev.40.0
+  - 2.0.0-dev.46.0
 
 env:
   - PKG=app TASK=dartanalyzer

--- a/Dockerfile
+++ b/Dockerfile
@@ -1,5 +1,5 @@
 # Keep version in-sync with .travis.yml and app/lib/shared/versions.dart
-FROM google/dart-runtime-base:2.0.0-dev.40.0
+FROM google/dart-runtime-base:2.0.0-dev.46.0
 
 # `apt-mark hold dart` ensures that Dart is not upgraded with the other packages
 #   We want to make sure SDK upgrades are explicit.

--- a/app/.travis.yml
+++ b/app/.travis.yml
@@ -2,7 +2,7 @@ language: dart
 
 dart:
   # Keep version in-sync with Dockerfile
-  - 2.0.0-dev.40.0
+  - 2.0.0-dev.46.0
 
 dart_task:
   - test

--- a/app/lib/shared/versions.dart
+++ b/app/lib/shared/versions.dart
@@ -7,23 +7,23 @@ import 'package:pub_semver/pub_semver.dart';
 import 'utils.dart' show isNewer;
 
 // update this whenever one of the other versions change
-final String runtimeVersion = '2018.3.23';
+final String runtimeVersion = '2018.4.7';
 final Version semanticRuntimeVersion = new Version.parse(runtimeVersion);
 
 // keep in-sync with SDK version in .travis.yml and Dockerfile
-final String sdkVersion = '2.0.0-dev.40.0';
+final String sdkVersion = '2.0.0-dev.46.0';
 final Version semanticSdkVersion = new Version.parse(sdkVersion);
 
 // keep in-sync with app/pubspec.yaml
-final String panaVersion = '0.10.5';
+final String panaVersion = '0.10.6';
 final Version semanticPanaVersion = new Version.parse(panaVersion);
 
 // keep in-sync with app/script/setup-flutter.sh
-final String flutterVersion = '0.2.4';
+final String flutterVersion = '0.2.9';
 final Version semanticFlutterVersion = new Version.parse(flutterVersion);
 
 // keep in-sync with SDK version in .travis.yml and Dockerfile
-final String dartdocVersion = '0.17.1+1';
+final String dartdocVersion = '0.18.1';
 final Version semanticDartdocVersion = new Version.parse(dartdocVersion);
 
 /// The version of our customization going into the output of the dartdoc static
@@ -33,7 +33,7 @@ final Version semanticCustomizationVersion =
     new Version.parse(customizationVersion);
 
 // Version that control the dartdoc serving.
-final dartdocServingRuntime = new Version.parse('2018.3.23');
+final dartdocServingRuntime = new Version.parse('2018.4.7');
 
 // Version that marks the default runtime version for analyzer entries created
 // before the runtime version was tracked.

--- a/app/pubspec.yaml
+++ b/app/pubspec.yaml
@@ -32,7 +32,7 @@ dependencies:
   yaml: '^2.1.12'
   # pana version to be pinned
   # keep in-sync with app/lib/shared/versions.dart
-  pana: '0.10.5'
+  pana: '0.10.6'
   # 3rd-party packages with pinned versions
   archive: '1.0.33'
   uuid: '0.5.3'

--- a/app/script/setup-flutter.sh
+++ b/app/script/setup-flutter.sh
@@ -22,7 +22,7 @@ then
   exit 1
 fi
 
-git clone -b v0.2.4 --single-branch https://github.com/flutter/flutter.git $FLUTTER_SDK
+git clone -b v0.2.9 --single-branch https://github.com/flutter/flutter.git $FLUTTER_SDK
 
 # Keep in-sync with app/lib/shared/versions.dart
 cd $FLUTTER_SDK

--- a/app/test/shared/versions_test.dart
+++ b/app/test/shared/versions_test.dart
@@ -19,7 +19,7 @@ void main() {
       dartdocVersion,
       customizationVersion,
     ].join('//').hashCode;
-    expect(hash, 610463204);
+    expect(hash, 927882828);
   });
 
   test('sdk version should match travis and dockerfile', () async {

